streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorContextImpl.java

```diff
@@ -217,12 +217,12 @@ public StateStore getStateStore(final String name) {
                     forward((ProcessorNode<K, V, ?, ?>) child, key, value);
                 }
             } else {
-                final ProcessorNode<K, V, ?, ?> child = currentNode().getChild(sendTo);
```

Comment:
Why change the `K, V` to `?, ?` -- and if we don't do that, do we still need the cast in the `forward` call down below?

Reply:
+1

Reply:
I made the change in ProcessorNode to add back the template types: https://github.com/apache/kafka/pull/9083/files/82b6f6f5d238401097e0906c8135c5c189524666#diff-705bfd0ed3f214048b76d775708cc7d2L96

But since `currentNode()`'s template is `<?, ?, ?, ?>` its templated `getChild` and that's why I need to weaken it here --- as you can see from the above `if` branch, it now aligns consistently on the typing.